File: web/src/types/InlineParserState.ts
import type Delimiter from "./Delimiter";
import type FootnoteReference from "./FootnoteReference";
import type InlineRule from "./InlineRule";
import type LinkReference from "./LinkReference";

export default interface InlineParserState {
	rules: Map<string, InlineRule>;

	src: string;
	i: number;
	line: number;
	lineStart: number;
	indent: number;
	delimiters: Delimiter[];
	refs: Record<string, LinkReference>;
	footnotes: Record<string, FootnoteReference>;

	// HACK:
	debug?: boolean;
}

File: dotnet/Allmark/Types/InlineParserState.cs
namespace Allmark.Types;

/// <summary>
/// Represents the state of the inline parser during markdown parsing.
/// </summary>
public record InlineParserState
{
	public required Dictionary<string, InlineRule> Rules { get; set; }

	public required string Src { get; set; }
	public required int I { get; set; }
	public required int Line { get; set; }
	public required int LineStart { get; set; }
	public required int Indent { get; set; }
	public required List<Delimiter> Delimiters { get; set; }
	public required Dictionary<string, LinkReference> Refs { get; set; }
	public required Dictionary<string, FootnoteReference> Footnotes { get; set; }

	// HACK:
	public bool? Debug { get; set; }
}
